src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -1768,7 +1768,7 @@ internal static string GetDataContractNamespaceFromUri(string uriString)
 
         private static string GetNamespacesDigest(string namespaces)
         {
-            byte[] namespaceBytes = Encoding.UTF8.GetBytes(namespaces);
+            byte[] namespaceBytes = DataContractSerializer.SafeUTF8.GetBytes(namespaces);
```

Comment:
This will have a bit more overhead than just using `Encoding.UTF8.GetBytes`.  What's the motivation for switching away from it?

Reply:
I'll revert it, out of curiosity is this slower because Encoding.UTF8 uses that sealed derived type?

Reply:
this is fixed in latest iteration but still curious about the question above https://github.com/dotnet/runtime/pull/76366#discussion_r983741441

Reply:
> is this slower because Encoding.UTF8 uses that sealed derived type?

The base UTF8Encoding's methods need to handle all possible configurations (e.g. encoding/decoding fallbacks).  The derived type that's used for Encoding.UTF8 only has a single configuration and overrides relevant methods to be optimized for that one configuration.  This makes them faster but also small enough to be inlined in some cases.  You can see this with a little benchmark like:
```C#
private static readonly UTF8Encoding s_base = new UTF8Encoding(true);
private static readonly MyUTF8 s_sealedDerived = new MyUTF8();

private sealed class MyUTF8 : UTF8Encoding
{
    public MyUTF8() : base(true) { }
}

[Benchmark(Baseline = true)]
public int CountBase() => s_base.GetMaxByteCount(5);

[Benchmark]
public int CountDerived() => s_sealedDerived.GetMaxByteCount(5);

[Benchmark]
public int CountUTF8() => Encoding.UTF8.GetMaxByteCount(5);
```

|       Method |      Mean |     Error |    StdDev | Ratio | RatioSD | Code Size |
|------------- |----------:|----------:|----------:|------:|--------:|----------:|
|    CountBase | 1.4344 ns | 0.0574 ns | 0.0661 ns |  1.00 |    0.00 |     349 B |
| CountDerived | 1.6461 ns | 0.0447 ns | 0.0418 ns |  1.14 |    0.04 |     349 B |
|    CountUTF8 | 0.5368 ns | 0.0408 ns | 0.0419 ns |  0.37 |    0.04 |      56 B |